### PR TITLE
Rework page to clarify how support access works

### PR DESCRIPTION
**Pull request by norosa**
Based on recent feedback, this page was causing some confusion about how support access works. Specifically, the purpose of the management role and the reason it requires some limited write permissions was not clear.

This pull request attempts to address those issues by:

- Adding a new section named *Technical details* which provides the full YAML configuration in a code block and includes a description of the management role and the reason it requires some limited write permissions.

- Reworking the rest of the document for clarity and style.

### CODE REVIEW

**Comment by norosa** [comment]
@leepa here's my redrafting of the *Read-Only Access to AWS* page, based on our discussions and the resources you provided me with (thx so much!)

while reworking the document, I tried my best to make sure the details/explanations were technically accurate. but there is a very good chance I have made a few mistakes here and there in my attempts to reword things for clarity (or where I thought it would be useful to expand things). please do scrutinize my work for errors

there's one thing I'm stuck on:

you provided me with a link to some internal source code that sets the maintenance policy attached to the support role. I would like to include the policy information in the document so it is as clear as the YAML file above, but I am not sure how to do that

I am presuming we don't want to embed the code that sets the policy. can you think of another way of sharing this policy info? ideally in the form of a literal block (like the YAML file)

File: docs/tickets/aws-ro-access.md [hunk 1]
```diff
@@ -1,27 +1,135 @@
...
+Outputs:
+  RoleName:
+    Value: !Ref AssumeableRole
+  RoleArn:
+    Value: !GetAtt AssumeableRole.Arn
+```
+
+Please note:
+
+* CloudFormation will create a management role that is responsible for creating, updating, and deleting support roles. For each support role, the management role uses generates a unique cryptographic key pair.
```
Comment (leepa) [inline docs/tickets/aws-ro-access.md:126]: This sentence does not make sense. Specifically `the management role uses generates a`. Maybe something like `our system uses the management role to apply a unique cryptographic key pair to the account`? Unless I miss what you're getting at with this sentence?

File: docs/tickets/aws-ro-access.md [hunk 2]
```diff
@@ -1,27 +1,135 @@
...
+
+  Because key creation has to be done by your [IdP](https://docs.aws.amazon.com/singlesignon/latest/userguide/idp.html), the management role itself requires some limited write permissions. However:
 
-Before deploying the AWS CloudFormation stack, you can download and review it at [https://doit-aws-ops-prod-templates.s3.amazonaws.com/doitmanagement.yaml](https://doit-aws-ops-prod-templates.s3.amazonaws.com/doitmanagement.yaml)
+  * The CMP has privileged access to the management role, and this is how our systems interact with your account. No DoiT staff members have access to the management role.
```
Comment (leepa) [inline docs/tickets/aws-ro-access.md:130]: The problem with No followed by a positive is that some people will miss the No (I did on the first read-through). Consider `DoiT Staff Members has no access to the management role`

File: docs/tickets/aws-ro-access.md [hunk 3]
```diff
@@ -1,27 +1,135 @@
...
+  * The management role's write permissions are tightly scoped and only grant what is necessary to perform key generation and the management of support roles.
 
-The CloudFormation will create an IAM Policy and IAM Role the DoiT International engineers can use to access your AWS account. The effective policy attached to the role is [SecurityAudit](https://console.aws.amazon.com/iam/home#policies/arn:aws:iam::aws:policy/SecurityAudit).
+* CloudFormation will create an IAM Policy and IAM Role for each staff member with the [SecurityAudit](https://console.aws.amazon.com/iam/home#policies/arn:aws:iam::aws:policy/SecurityAudit) policy attached to each role.
```
Comment (leepa) [inline docs/tickets/aws-ro-access.md:133]: This is not what the CfN does.

File: docs/tickets/aws-ro-access.md [hunk 4]
```diff
@@ -1,27 +1,135 @@
...
 
-The CloudFormation will use a small AWS Lambda function to notify our system of the created role so that we can add it to our system automatically, this saves you copying and pasting. You can see in the CloudFormation that the Lambda runs with very restricted permissions to just serve this purpose.
+* CloudFormation uses a small AWS Lambda function (with very restricted permissions) to pass key information back to the CMP when necessary.
```
Comment (leepa) [inline docs/tickets/aws-ro-access.md:135]: This was removed in the newer CfN and so no Lambda function is used anymore.

File: docs/tickets/aws-ro-access.md (hunk 1 above)
Comment (norosa) [reply docs/tickets/aws-ro-access.md:126]: no good clarification thanks!

File: docs/tickets/aws-ro-access.md (hunk 2 above)
Comment (norosa) [reply docs/tickets/aws-ro-access.md:130]: good catch

File: docs/tickets/aws-ro-access.md (hunk 3 above)
Comment (norosa) [reply docs/tickets/aws-ro-access.md:133]: fixed to "we" (as I now realize that this is done directly by us using the management role)

(please correct if wrong once I've pushed my fixup!)

File: docs/tickets/aws-ro-access.md (hunk 4 above)
Comment (norosa) [reply docs/tickets/aws-ro-access.md:135]: removed, thx

**Comment by norosa** [comment]
@leepa pushed fixups for your comments. once you let me know what to do about the user roles/perms, I'll make another fixup commit

File: docs/tickets/aws-ro-access.md [hunk 5]
```diff
@@ -1,27 +1,133 @@
 ---
-description: how to securely grant DoiT International access to your AWS account
+description: How to grant Doit International read-only support access to your AWS account
```
Comment (samrogowicz) [inline docs/tickets/aws-ro-access.md:2]: This line has `Doit`, but ln. 7 has `DoiT` -> I gather we should be using `DoiT` in all places?

Comment (norosa) [inline docs/tickets/aws-ro-access.md:2]:
```suggestion
description: How to grant DoiT International read-only support access to your AWS account
```

**Review by leepa — APPROVED** [review]

**Pull request merged**